src/bin/idl/mod.rs

```diff
@@ -332,6 +320,18 @@ fn instruction(
     Ok(())
 }
 
+fn mutable_account_exists(accounts: &[IdlAccountItem]) -> bool {
+    let mut exists = false;
+    for item in accounts {
+        exists |= match item {
+            IdlAccountItem::IdlAccount(acc) => acc.is_mut,
+            IdlAccountItem::IdlAccounts(accs) => mutable_account_exists(&accs.accounts),
+        }
+    }
+
+    exists
```

Comment:
Do you not like iterators?
```suggestion
    accounts.iter().any(|item| match item {
            IdlAccountItem::IdlAccount(acc) => acc.is_mut,
            IdlAccountItem::IdlAccounts(accs) => mutable_account_exists(&accs.accounts),
        })
```

Reply:
> Why do you not like iterators?

I am not used to them. I am used to writing `for(int I=0; i<n; i++)` in C, so I often write for-loops in Rust too.